src/plugins/application_config/README.md

```diff
@@ -61,13 +61,13 @@ Let's call this plugin `MyConfigurationClientPlugin`.
 First, this plugin will need to implement a class `MyConfigurationClient` based on interface `ConfigurationClient` defined in the `types.ts` under directory `src/plugins/application_config/server/types.ts`. Below are the functions inside the interface.
 
 ```
-  getConfig(): Promise<Map<string, string>>;
+  getConfig(options?: ConfigurationClientOptions): Promise<Map<string, string>>;
 
-  getEntityConfig(entity: string): Promise<string>;
+  getEntityConfig(entity: string, options?: ConfigurationClientOptions): Promise<string>;
```

Comment:
Nit:
Thanks for this change, this comment is not a blocker of this PR~

Since `entity` also get passed from `Request`. What if we also put `entity` as an attribute of `ConfigurationClientOptions` and only pass `ConfigurationClientOptions` into these functions ?

Reply:
There is a subtle difference here. The parameter `options?:` has a `?` which indicates an optional parameter. As a contrast, `entity` is a mandatory parameter.